src/ast.ls

```diff
@@ -1939,30 +1955,28 @@ class exports.Fun extends Node
             params.0 = Var \it if body.traverse-children -> it.value is \it or null
         names   = []
         assigns = []
-        if params.length
-            dic = {}
-            for p in params
-                vr = p
-                vr.=first if df = vr.get-default!
-                if vr.is-empty!
-                    vr = Var scope.temporary \arg
-                else if vr.value is \..
-                    vr = Var o.ref = scope.temporary!
-                else if vr not instanceof Var
-                    unaries = []
-                    while vr instanceof Unary
-                        has-unary = true
-                        unaries.push vr
-                        vr.=it
-                    v = Var delete (vr.it || vr)name || vr.var-name! || scope.temporary \arg
-                    assigns.push Assign vr, switch
-                        | df        => Binary p.op, v, p.second
-                        | has-unary => fold ((x, y) -> y.it = x; y), v, unaries.reverse!
-                        | otherwise => v
-                    vr = v
-                else if df
-                    assigns.push Assign vr, p.second, \=, p.op, true
-                names.push (scope.add vr.value, \arg, p), ', '
+        for p in params
+            vr = p
```

Comment:
if you want to use this occasion to give more meaningful names to the variables around, feel very free to...